src/common/streak.ts

```diff
@@ -0,0 +1,75 @@
+import { DayOfWeek, DEFAULT_WEEK_START } from './date';
+import {
+  Day,
+  FREEZE_DAYS_IN_A_WEEK,
+  STREAK_RECOVERY_MAX_GAP_DAYS,
+} from './users';
+
+interface DaysProps {
+  day: number;
+  firstDayOfWeek: Day;
+  lastDayOfWeek: Day;
+  lastView: Date;
+}
+
+const getNextDay = (day: number) => (day === 6 ? 0 : day + 1);
+const getPreviousDay = (day: number) => (day === 0 ? 6 : day - 1);
```

Comment:
There are functions in date-fns to get previous and next day from date and then you can get the number of the day from that with `getDay` as usual, feels more bulletproof then doing this helpers.

Reply:
Their implementation of that is a bit different from our requirements. They need a Date to tell the next day, for us, we just need to get the next day if the day today is Saturday (6).

Reply:
I see you just want to iteratate 0-6 without overflow, you can also do `(new Date().getDay() + 1) % 7` which will go to 0 when day is 6. More typical then doing a exact `=== 6` check.

Reply:
Ah, right. Yeah, we can do that.

Reply:
Updated the functions now.